pexpect/expect.py

```diff
@@ -27,11 +28,11 @@ def new_data(self, data):
             window = spawn.buffer
         index = searcher.search(window, len(data))
         if index >= 0:
-            value = spawn.buffer
             spawn._buffer = spawn.buffer_type()
-            spawn._buffer.write(value[searcher.end:])
-            spawn.before = value[: searcher.start]
-            spawn.after = value[searcher.start: searcher.end]
+            spawn._buffer.write(window[searcher.end:])
```

Comment:
@takluyver these search match indexes are _actually_ relative to `window`, which _may not_ be equal to `spawn.buffer` when `searchwindowsize` is in use.